pyfarm/agent/http/api/state.py

```diff
@@ -46,6 +53,15 @@ def post(self, **kwargs):
         request = kwargs["request"]
         data = kwargs["data"]
         agent = config["agent"]
+
+        try:
+            os.remove(config["run_control_file"])
+        except (WindowsError, OSError, IOError) as e:
+            if e.errno != ENOENT:
+                logger.error("Could not delete run control file %s: %s: %s",
+                             config["run_control_file"],
+                             type(e).__name__, e)
```

Comment:
At least on Windows I've seen the behavior several times where os.remove doesn't work because Windows still considers the file locked for some reason.  Usually, calling os.remove as the interpreter exits will clean it up.